addons/pos_self_order/models/product_product.py

```diff
@@ -97,16 +97,33 @@ def _get_price_info(
                 self, qty, currency=pos_config.currency_id
             )
 
-        price_info = pos_config.default_fiscal_position_id.map_tax(self.taxes_id).compute_all(
+        display_price_default = price
+        display_price_alternative = price
+
+        taxes_default = pos_config.default_fiscal_position_id.map_tax(self.taxes_id)
+        taxes_alternative = pos_config.self_ordering_alternative_fp_id.map_tax(self.taxes_id)
+
+        all_prices_default = taxes_default.compute_all(
+            price, pos_config.currency_id, qty, product=self
```

Comment:
Won't change the result, but might be clearer to use display_price_default here and displau_price_alternative for the next.

Reply:
Oh, variables above are just declaration. The real setting of their values is below.

Reply:
I added a comment above the declaration. I should not change the variable passed in the function. It should be clear that it's the `price` passed in the function or calculated above.